hangman_game.py

```diff
@@ -0,0 +1,161 @@
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 2:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print('  |   /|\ ')
+            print('  |      ')
+            print('__|__\n')
+        if self.hangmanState == 1:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print("  |   /|\ ")
+            print('  |   / ' )
+            print('__|__\n')
+        if self.hangmanState <= 0:
+            print('  |‾‾‾‾|')
+            print('  |    0')
+            print("  |   /|\ ")
+            print("  |   / \ ")
+            print('__|__\n')
+
+    def showSplit(self, array):
+        print(' '.join(array))  # format output for hangman
+
+    def playGame(self, minLength):
+        import numpy as np
+        target = self.setupGame(minLength)  # specify target word with setupGame
+        targetSplit = np.array(list(target))  # created for finding correct guesses
+        maskedSplit = np.array(['_' for letter in list(target)])  # created to show state of hangman game
+
+        while self.hangmanState > 0:
+            self.showBoard()  # start off by showing state of hangman
+            self.showSplit(maskedSplit)  # display state of board
+            # print(target)  # used for debugging
+            letterGuess = input('Guess a letter ')
+            if letterGuess.upper() in self.validLetters:
+                if letterGuess.upper() in self.guessedLetters:
+                    self.hangmanState -= 1  # decrement for repeated guess
+                    print('Already guessed this letter')
+                else:
+                    if letterGuess.upper() in targetSplit:
+                        validSpots = np.where(targetSplit == letterGuess.upper())  # all indices of match
+                        self.guessedLetters.append(letterGuess.upper())  # add guess to list of guesses
+                        for index in validSpots:
+                            maskedSplit[index] = letterGuess.upper()  # replace masked spot with letter
+                        if '_' not in maskedSplit:
+                            print('You won the game!')
+                            self.showSplit(maskedSplit)  # show the completed word
+                            self.showBoard()
+                            break  # exit the game since word was correctly guessed
+                    else:
+                        self.hangmanState -= 1  # decrement hangman state for incorrect guess
+                        print('Incorrect guess')
+                        self.guessedLetters.append(letterGuess.upper())
+            else:
+                print('Invalid input, guess a letter')
+        else:
+            print('Game Over!')
+            self.showSplit(maskedSplit)  # show the masked split
+            self.showBoard()  #  show the board
+            print('Correct word was {}'.format(target))  # tell user what correct word was
+
+
+
+def main():
```

Comment:
the code at the top (the argparse stuff should be here in main). We dont necessarilly want that run unless we are running main right?